zerver/decorator.py

```diff
@@ -352,7 +352,7 @@ def _wrapped_func_arguments(request, *args, **kwargs):
                 # case insensitive per RFC 1945
                 if auth_type.lower() != "basic":
                     return json_error(_("Only Basic authentication is supported."))
-                role, api_key = base64.b64decode(encoded_value).split(":")
+                role, api_key = base64.b64decode(encoded_value.encode('utf-8')).decode('utf-8').split(":")
```

Comment:
I should have used `force_bytes` here, since `request.META`'s value type is probably `str`, not `text_type`. I found this out by experiment while annotating zerver/views/webhooks/beanstalk.py.